validation-test/stdlib/Data.swift

```diff
@@ -0,0 +1,32 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+// REQUIRES: objc_interop
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+import Foundation
+
+var DataTestSuite = TestSuite("Data")
+DataTestSuite.test("Data.Iterator semantics") {
+  // Empty data
+  checkSequence(Data(), [])
+
+  // Small data
+  checkSequence(Data(bytes: [1,2,4,8,16]), [1,2,4,8,16])
+
+  // Boundary conditions
+  checkSequence(Data(bytes: [5]), [5])
+  checkSequence(Data(bytes: Array(1...31)), 1...31)
+  checkSequence(Data(bytes: Array(1...32)), 1...32)
+  checkSequence(Data(bytes: Array(1...33)), 1...33)
+
+  // Large data
+  var data = Data(count: 65535)
+  data.withUnsafeMutableBytes { (ptr: UnsafeMutablePointer<UInt8>) -> () in
+    for i in 0..<data.count {
+      ptr[i] = UInt8(i % 23)
+    }
+  }
+  checkSequence(data, (0..<65535).lazy.map({ UInt8($0 % 23) }))
```

Comment:
One more thing: the expected value is the first parameter, sequence under test is the second.

```
public func checkSequence<
  ${genericParam}, S : Sequence
>(
  _ expected: ${Expected},
  _ sequence: S,
```

We should probably add argument labels to this function.